lib/datadog/ci/transport/remote_settings_api.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+require "json"
+
+require "datadog/core/environment/identity"
+
+require_relative "../ext/transport"
+
+module Datadog
+  module CI
+    module Transport
+      # Datadog API client
+      # Calls settings endpoint to fetch library settings for given service and env
+      class RemoteSettingsApi
+        class Response
+          def initialize(http_response)
+            @http_response = http_response
+            @json = nil
+          end
+
+          def ok?
+            resp = @http_response
+            !resp.nil? && resp.ok?
+          end
+
+          def payload
+            cached = @json
+            return cached unless cached.nil?
+
+            resp = @http_response
+            return @json = default_payload if resp.nil? || !resp.ok?
+
+            begin
+              @json = JSON.parse(resp.payload).dig(*Ext::Transport::DD_API_SETTINGS_RESPONSE_DIG_KEYS) ||
+                default_payload
+            rescue JSON::ParserError => e
+              Datadog.logger.error("Failed to parse settings response payload: #{e}. Payload was: #{resp.payload}")
+              @json = default_payload
+            end
+          end
+
+          private
+
+          def default_payload
+            Ext::Transport::DD_API_SETTINGS_RESPONSE_DEFAULT
+          end
+        end
+
+        def initialize(api: nil, dd_env: nil)
+          @api = api
+          @dd_env = dd_env
+        end
+
+        def fetch_library_settings(test_session)
+          api = @api
+          return Response.new(nil) unless api
+
+          request_payload = payload(test_session)
+          Datadog.logger.debug("Fetching library settings with request: #{request_payload}")
+
+          http_response = api.api_request(
+            path: Ext::Transport::DD_API_SETTINGS_PATH,
+            payload: request_payload
+          )
+
+          Datadog.logger.debug("Library settings response: #{http_response}")
+
+          Response.new(http_response)
+        end
+
+        private
+
+        def payload(test_session)
+          {
+            "data" => {
+              "id" => Datadog::Core::Environment::Identity.id,
+              "type" => Ext::Transport::DD_API_SETTINGS_TYPE,
+              "attributes" => {
+                "service" => test_session.service,
```

Comment:
I think including the `test_level` here is a requirement to take advantage of the `require_git` optimization for ITR (which will allow you to not always run with code coverage). Unless you're going to switch between suite and test skipping, you probably can set it to `test`

Reply:
same question as above: where is test_level specified?